### ISSUE TITLE
Problem with intersection type

### ISSUE THREAD

**Issue by koko236** [facebook/flow#4374]
I have following types defined:
```
export type MovementInfo = { id: string, x: number, y: number, s: [number, number] }
export type TypeInfo = { eType: string, idx: number }
export type CreationInfo = TypeInfo & MovementInfo
export type CmdNew = { type: 'new' } & CreationInfo
```
In VSCode Flow underlines both `TypeInfo` and `MovementInfo` in 3rd line and `{ type: 'new' }` in 
 4th and gives me following errors for each:
```
[flow] object type (This type is incompatible with an implicitly-returned undefined.)
```
Using Flow 0.49.1
Is it me or Flow doing something stupid?

**Comment by LoganBarnett** [comment]
From the looks of it, we're missing some greater context. Putting your code in the [online flow checker](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVBTAHgBzgJwBcxCBPHDMAWTgDcMBbDAO0IElmo4wBeMAbzABLACYAuMAGdC+IcwDmAGjBYJzAK4MARhnzLSazTr1SJAbQ3bdyy8YC6YAL6ZcBYmQpgAKuQwcuvAJgGD4UEtKyCsqiqmC2uk4ueEQkvmAAwvgYAIaEQnDM-tx8oX6c3ABk1HSMLOzlSW6pnukMIgByGEh8gh4YEgDkzF0DTmBVmTl5BUVAA) passes. Can you provide a https://flow.org/try link to a reproduction of the error?

**Comment by koko236** [comment]
Hi, I got similar suggestion on SO that it's probably some function using
these types elsewhere in code causing the error here. The problem is I had
Flow working inconsistently for some time and I managed to use these in a
lot of places without a sign of trouble. I started getting these errors
after I updated flow-bin and now I'll need some time to chase the culprit.
Thanks anyway for confirming the code itself is ok I was tearing my hair
out already. :-)

Konrad Grzegorzewicz
t: +48 502 404 903
